packages/js/src/core/fetch-token.ts

```diff
@@ -0,0 +1,75 @@
+import { KeysToCamelCase } from '@silverhand/essentials';
+import { string, number, assert, type, StructError, Infer } from 'superstruct';
+import { Except } from 'type-fest';
+
+import { LogtoError } from '../utils/errors';
+import { createRequester } from '../utils/requester';
+
+export interface FetchTokenByRefreshTokenParameters {
+  clientId: string;
+  tokenEndPoint: string;
+  refreshToken: string;
+  resource?: string;
+  scope?: string[];
+}
+
+const TokenResponseSchema = type({
+  access_token: string(),
+  refresh_token: string(),
+  id_token: string(),
+  scope: string(),
+  expires_in: number(),
+});
+
+export type RefreshTokenTokenResponse = Except<
+  KeysToCamelCase<Infer<typeof TokenResponseSchema>>,
+  'scope'
+> & {
+  scope: string[];
+};
+
+export const fetchTokenByRefreshToken = async ({
+  clientId,
+  tokenEndPoint,
+  refreshToken,
+  resource,
+  scope,
+}: FetchTokenByRefreshTokenParameters): Promise<RefreshTokenTokenResponse> => {
+  const parameters = new URLSearchParams();
+  parameters.append('client_id', clientId);
+  parameters.append('refresh_token', refreshToken);
+  if (resource) {
```

Comment:
empty line before this line?

Reply:
OK, let me take a look at the linting rules

Reply:
Done.